controllers/user/Notification.js

```diff
@@ -0,0 +1,107 @@
+import moment from 'moment';
+import db from '../../models/index';
+import Util from '../../helpers/Util';
+import HttpError from '../../helpers/exceptionHandler/httpError';
+
+const { Notifications } = db;
+/**
+ * @exports Notification
+ * @class Notification
+ * @description Handles all notifications in the application
+ * */
+class Notification { /* eslint-disable */
+
+    constructor() {
+        this.message;
+        this.userId;
+    }
+
+
+    /**
+     * get all the notification of a user
+     * @async
+     * @param  {object} req - Request object
+     * @param {object} res - Response object
+     * @return {json} Returns json object
+     * @static
+     */
+    async getNotification(req, res) {
+        const { userId } = req.authData;
+        const total = await Notifications.count({ where: { userId } });
+        const pageInfo = Util.getPageInfo(req.query.page, req.query.size, total);
+        const { page, limit, offset } = pageInfo;
+        let userNotification = await Notifications.findAll({
+            limit,
+            offset,
+            where: { userId },
+            order: [
+                ['createdAt', 'DESC']
+            ]
+        });
+
+        if (userNotification.length === 0) {
+            return res.status(200).json({ status: 'success', message: 'You do not have any notification yet' });
+        }
+        userNotification = userNotification.map((notification) => {
+            const {
+                id,
+                isRead,
+                message,
+                createdAt,
+                updatedAt
+            } = notification;
+            const currentTime = moment(Date.now());
+            if (isRead && moment(updatedAt).diff(currentTime, 'days')) {
+                Notifications.destroy({ where: { id } });
```

Comment:
@blackshady , it would be nice if you could separate concerns here. A `getNotifications()` function should `only` get notifications and maybe use the event emitter pattern to trigger a `destroyNotification()` function as a side effect. Better still, we could have a scheduler run a `bulkDelete` script for read notifications every week or so.